Stream_name: Adding a default stream name to the Stream Info.

PawBud opened the pull request:
This PR aims to fix https://bugs.launchpad.net/mixxx/+bug/1818054.

Discussion on Zulip: https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/adding.20a.20default.20Stream.20Name.

PawBud committed:
Stream_name: Adding a default stream name to the Stream Info.

Previously, the default stream name was blank. To accomodate consistency,
this commit will add 'Mixxx' to the stream name.

ronso0 commented on src/preferences/broadcastprofile.cpp:
```diff
@@ -76,6 +76,8 @@ const bool kDefaultNoDelayFirstReconnect = true;
 const bool kDefaultOggDynamicupdate = false;
 double kDefaultReconnectFirstDelay = 0.0;
 double kDefaultReconnectPeriod = 5.0;
+const QString kDefaultStreamName =
+        QObject::tr("Mixxx");
```
```suggestion
const QString kDefaultStreamName = QStringLiteral("Mixxx");
```
As @Holzhaus said: Mixxx is not going to be translated, you can use QStringLiteral instead.

PawBud replied:
I see so that's what the QStringLiteral meant!

ronso0 commented:
This is a harmless bug fix so you can target mixxx:2.3 instead of master.

There shouldn't be a need to rebase, but in case you see unrelated commits poping up here after you switched to mixxx:2.3, just run `git rebase -i XXX/2.3` where XXX is what you named the mixxxdj/mixxx remote.

ronso0 commented:
Unfortunately users that upgraded Mixxx can't benefit from this fix instantly due to the following constraints (not caused by this branch):
1. I expected "Create new connection" in Pref > Broadcast to create a new connection from the default profile, but it just copies the existing connection. That was discussed only briefly in https://github.com/mixxxdj/mixxx/pull/1300#issuecomment-317378535.
As @Palakis mentioned, we shoul dhave `Clone (selected) connection` and `Create new (default) connection`
2. Reset to defaults does nothing (on purpose), but this too inhibits getting a fresh default connection
3. last connection cannot be deleted

All in all, users can not get a default connection again after altering the initial settings unless they remove all connection files from `$CONFIG_DIR/broadcast_profiles` manually.

PawBud committed:
Stream_name: Replaced QString with QStringLiteral.

This commit makes it such that, The kDefaultStreamName need not be translated.

Co-authored-by: ronso0 <ronso0@mixxx.org>

PawBud commented:
> Unfortunately users that upgraded Mixxx can't benefit from this fix instantly due to the following constraints (not caused by this branch):
> 
> 1. I expected "Create new connection" in Pref > Broadcast to create a new connection from the default profile, but it just copies the existing connection. That was discussed only briefly in [#1300 (comment)](https://github.com/mixxxdj/mixxx/pull/1300#issuecomment-317378535).
>    As @Palakis mentioned, we shoul dhave `Clone (selected) connection` and `Create new (default) connection`
> 2. Reset to defaults does nothing (on purpose), but this too inhibits getting a fresh default connection
> 3. last connection cannot be deleted
> 
> All in all, users can not get a default connection again after altering the initial settings unless they remove all connection files from `$CONFIG_DIR/broadcast_profiles` manually.
> 
> **EDIT**
> 3. was already touch in #2158 bu not considered worth fixing

So, bottom line, do I have to wait until a Fix is issued, although you clearly stated that they r not really "bugs"

ronso0 commented:
never mind, as I wrote it's unrelated to your fix, just a heads-up for everyone reading this.
Users starting with a new Mixxx install (or clean config) will get the stream name filled.

> This is a harmless bug fix so you can target mixxx:2.3 instead of master.

This can be done at the top of this page, 'Edit', then select mixxx/2.3 in the drop-down.
 
> There shouldn't be a need to rebase, but in case you see unrelated commits poping up here after you switched to mixxx:2.3, just run `git rebase -i XXX/2.3` where XXX is what you named the mixxxdj/mixxx remote.

PawBud commented:
@ronso0 so the thing is I picked the commits (and didn't squash any), It should rebase but it's showing some conflicts, I tried to manually `Accept Incoming Changes` and merged them with the commit message as `Resolving Conflicts` in VSC, but it just kept on aborting. What am I doing wrong?

Holzhaus commented:
Please follow the instructions here: https://github.com/mixxxdj/mixxx/wiki/Using%20Git#targeting-another-base-branch

PawBud commented:
> Please follow the instructions here: https://github.com/mixxxdj/mixxx/wiki/Using%20Git#targeting-another-base-branch

just what I needed! Thanks a lot, I ll remember the command from now on I guess :).

PawBud commented:
@ronso0, It's weird that the default string is not showing up:-
![Screenshot from 2020-08-28 19-28-20](https://user-images.githubusercontent.com/48093317/91574402-c63b5500-e964-11ea-9619-701d06b2d06a.png)

ronso0 commented:
I guess that's exactly what I discovered [above](https://github.com/mixxxdj/mixxx/pull/3043#issuecomment-679268439)

Try removing all files from `$CONFIG_DIR/broadcast_profiles` (~/.mixxx/broadc.. on linux), or remove/backup the config dir entirely.
Then it shows up for the initial default new connection.

PawBud commented:
> I guess that's exactly what I discovered [above](https://github.com/mixxxdj/mixxx/pull/3043#issuecomment-679268439)
> 
> Try removing all files from `$CONFIG_DIR/broadcast_profiles` (~/.mixxx/broadc.. on linux), or remove/backup the config dir entirely.
> Then it shows up for the initial default new connection.

I can't really find the config directory, there is a file `mixxx\.tx\config`

ronso0 commented:
https://github.com/mixxxdj/mixxx/wiki/Finding%20the%20Mixxx.log%20file

PawBud commented:
![Screenshot from 2020-08-28 20-15-44](https://user-images.githubusercontent.com/48093317/91580819-45cc2280-e96b-11ea-8530-31154576d619.png)
![Screenshot from 2020-08-28 20-15-36](https://user-images.githubusercontent.com/48093317/91580822-46fd4f80-e96b-11ea-9a0a-b5b3f6783ae3.png)
It's pretty embarrassing to further ask your help, but I can't find any file named `mixxx.log`, moreover, I have verified that there are no Hidden dir.

Holzhaus commented:
The hidden `.mixxx` directory is in your home directory.

ronso0 commented:
Please read [Wiki: Finding the mixxx.log file](https://github.com/mixxxdj/mixxx/wiki/Finding%20the%20Mixxx.log%20file)
There's a `.mixxx` folder in you Home directory, like `home/YourUsername/.mixxx/`
Make sure you have 'Show hidden folder' enabled for your file manager.

ronso0 commented:
(I update the wiki page to include that info)

PawBud commented:
oh ok thanks a lot, should be easier for others now :)

ronso0 commented:
Any comments on the minimal changes? If not I'd merge this.

daschuer commented:
@PawBud: Thank you for this fist pull request.

Before merge we need you permission. 
Please sign: 
https://docs.google.com/a/mixxx.org/spreadsheet/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ
and comment here when done.

daschuer reviewed (approved):
LGTM.  (the CI failure is unrelated.)

PawBud commented:
Done @daschuer

daschuer commented:
Thank you.

daschuer commented:
https://github.com/mixxxdj/mixxx/commit/6c5bc4d43d2
